features/upgrade/monitoring/monitoring-upgrade.feature

```diff
@@ -0,0 +1,77 @@
+    Then Then the step should succeed
+    When I run the :get client command with:
+      | resource      | clusteroperator |
+      | resource_name | monitoring      |
+    And evaluation of `@result[:stdout].split(/\n/)[1].split(/\s+/)[4]` is stored in the :degreaded_status clipboard
+    Then the expression should be true> cb.degreaded_status == "False"
+
+    #check retention time
+    When I run the :get client command with:
+      | resource      | prometheus |
+      | resource_name | k8s        |
+      | o             | yaml       |
+    Then the expression should be true> YAML.load(@result[:stdout])["spec"]["retention"] == "3h"
+
+    # get sa/prometheus-k8s token
+    When evaluation of `secret(service_account('prometheus-k8s').get_secret_names.find {|s| s.match('token')}).token` is stored in the :sa_token clipboard
+
+    # curl -k -H "Authorization: Bearer $token" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=machine_cpu_cores'
+    When I run the :exec admin command with:
+      | n                | openshift-monitoring |
+      | pod              | prometheus-k8s-0     |
+      | c                | prometheus           |
+      | oc_opts_end      |                      |
+      | exec_command     | sh                   |
+      | exec_command_arg | -c                   |
+      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=machine_cpu_cores |
+    Then the step should succeed
+    And the output should contain:
+      | "__name__":"machine_cpu_cores" |
+
+    # curl -k -H "Authorization: Bearer $token" 'https://alertmanager-main.openshift-monitoring.svc:9094/api/v1/alerts'
+    When I run the :exec admin command with:
+      | n                | openshift-monitoring |
+      | pod              | prometheus-k8s-0     |
+      | c                | prometheus           |
+      | oc_opts_end      |                      |
+      | exec_command     | sh                   |
+      | exec_command_arg | -c                   |
+      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v1/alerts |
+    Then the step should succeed
+    And the output should contain:
+      | "alertname":"Watchdog" |
+
+    # curl -k -H "Authorization: Bearer $token" 'https://grafana.openshift-monitoring.svc:3000/api/health
+    When I run the :exec admin command with:
+      | n                | openshift-monitoring |
+      | pod              | prometheus-k8s-0     |
+      | c                | prometheus           |
+      | oc_opts_end      |                      |
+      | exec_command     | sh                   |
+      | exec_command_arg | -c                   |
+      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://grafana.openshift-monitoring.svc:3000/api/health |
```

Comment:
@lihongyan1 is this route exposed externally?  Is so we have a step to do query via an external route.

Reply:
yes, there is external route, but we can not query via the route, the results will not shown in the query, the results are hidden from the query due to the frame protection
`# token=`oc sa get-token prometheus-k8s -n openshift-monitoring`
# oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://alertmanager-main-openshift-monitoring.apps.qe-lpt-526.qe.devcluster.openshift.com/#/alerts'
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Alertmanager</title>
    </head>
    <body>
        <script>
            // If there is no trailing slash at the end of the path in the url,
            // add one. This ensures assets like script.js are loaded properly
            if (location.pathname.substr(-1) != '/') {
                location.pathname = location.pathname + '/';
                console.log('added slash');
            }
        </script>
        <script src="script.js"></script>
        <script>
            var app = Elm.Main.init({
                flags: {
                    production: true,
                    defaultCreator: localStorage.getItem('defaultCreator')
                }
            });
            app.ports.persistDefaultCreator.subscribe(function(name) {
                localStorage.setItem('defaultCreator', name);
            });
        </script>
    </body>
</html>
`

Reply:
@peter, please share the way to verify data from external route, thx
The corresponding external route of this service
https://grafana-openshift-monitoring.apps.hongyali-aws-4-26.qe.devcluster.openshift.com/api/health